src/app/shared/reports/calculations/assessmentReport.ts

```diff
@@ -63,30 +62,68 @@ export function getAssessmentReport(assessment: IdbAssessment, energyOpportuniti
         totalEnergyCostSavings += energyOpportunityCostSavings;
     };
 
+    if (assessment.costSavings) {
+        totalEnergyCostSavings += assessment.costSavings;
+    }
+
     let totalAssessmentNebSavings: number = _.sumBy(assessmentNebReports, (report: NebReport) => {
         return report.totalCostSavings
     });
     let energyOpportunityNebSavings: number = _.sumBy(energyOpportunityNebReports, (report: NebReport) => {
         return report.totalCostSavings
     });
-    let totalNebSavings: number = totalNonOpportunitySavings + energyOpportunityNebSavings;
+    let totalNebSavings: number = totalAssessmentNebSavings + energyOpportunityNebSavings;
 
     let totalCostSavings: number = totalEnergyCostSavings + totalNebSavings;
 
-    let totalEnergySavings: number = _.sumBy(energyOpportunityReports, (report: EnergyOpportunityReport) => {
-        return report.energyOpportunity.energySavings
-    }) + assessment.energySavings;
+    let opportunityEnergySavings: number = _.sumBy(energyOpportunityReports, (report: EnergyOpportunityReport) => {
+        if (report.energyOpportunity.energySavings) {
+            return report.energyOpportunity.energySavings;
+        }
+        return 0;
+    });
 
-    let implementationCost: number = _.sumBy(energyOpportunityReports, (report: EnergyOpportunityReport) => {
-        return report.energyOpportunity.implementationCost
-    }) + assessment.implementationCost
+    let totalEnergySavings: number = 0;
+    if (assessment.energySavings) {
+        totalEnergySavings += assessment.energySavings;
+    }
+    if (energyOpportunityCostSavings) {
```

Comment:
should it be:
  if (energyOpportunityCostSavings) {
        totalEnergySavings += energyOpportunityCostSavings;
    };
OR
  if (opportunityEnergySavings) {
        totalEnergySavings += opportunityEnergySavings;
    };

Reply:
good catch. Fixed